components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/template/service/BallerinaOperation.java

```diff
@@ -129,6 +129,13 @@ public BallerinaOperation buildContext(Operation operation, ExtendedAPI api) thr
         this.scope = OpenAPICodegenUtils.getMgwResourceScope(operation);
         //set resource level endpoint configuration
         setEpConfigDTO(operation);
+        //If production/sandbox endpoints for both API and Operation is not available, Error should be thrown.
+        if (api.getEndpointConfigRepresentation().getProdEndpointList() == null &&
+                api.getEndpointConfigRepresentation().getSandboxEndpointList() == null
+                && !hasProdEpConfig
+                && !hasSandEpConfig) {
+            throw new CLICompileTimeException("The endpoint Configuration is not provided for the operation.");
```

Comment:
Better to mention which operation this is. Otherwise user have to go through all operations to find which operation is causing the issue.

Reply:
This was implemented in such a way that the exception thrown in here is caught from the method above. Otherwise, we need to pass the operation name, path information to this layer. This is a sample error message.
```
micro-gw: Error while parsing information under path:/products/{productId}in the API "Online-Store-Api:v1.0.0".
        -Error while parsing the information under GET resource.
        -The endpoint Configuration is not provided for the operation.
```

Reply:
Great